core/crypto/secp256k1_types.hpp

```diff
@@ -13,23 +13,39 @@ namespace kagome::crypto::secp256k1 {
     static constexpr size_t kUncompressedPublicKeySize = 65u;
     static constexpr size_t kCompressedPublicKeySize = 33u;
     static constexpr size_t kCompactSignatureSize = 65u;
+    static constexpr size_t kTruncatedPublicKeySize = 64u;
   }  // namespace constants
 
   using EcdsaVerifyError = uint8_t;
   namespace ecdsa_verify_error {
     static constexpr EcdsaVerifyError kInvalidRS = 0;
     static constexpr EcdsaVerifyError kInvalidV = 1;
     static constexpr EcdsaVerifyError kInvalidSignature = 2;
+    static constexpr EcdsaVerifyError kNoError =
+        std::numeric_limits<EcdsaVerifyError>::max();
   }  // namespace ecdsa_verify_error
 
+  /**
+   * @brief converts ourcome::result failure to
+   * @param failure outcome::result containing error
+   * @return error code
+   */
+  EcdsaVerifyError convertFailureToError(outcome::result<void> failure);
+
   /**
    * compressed form of public key
    */
   using CompressedPublicKey = common::Blob<constants::kCompressedPublicKeySize>;
+
   /**
    * uncompressed form of public key
    */
-  using ExpandedPublicKey = common::Blob<constants::kUncompressedPublicKeySize>;
+  using UncompressedPublicKey = common::Blob<constants::kUncompressedPublicKeySize>;
+
+  /**
+   * truncated form of uncompressed public key
+   */
+  using TruncatedPublicKey = common::Blob<constants::kTruncatedPublicKeySize>;
```

Comment:
I guess it inapprotriate name, because uncompressed public key without tag is full or origin public key.

Reply:
I could approve with comment.